File: Greenhouse-backend/src/main/java/dev/canercin/greenhouseiot/dto/GreenhouseWeeklyAverageDTO.java
package dev.canercin.greenhouseiot.dto;

import lombok.Data;
import java.time.Year;

@Data
public class GreenhouseWeeklyAverageDTO {
    private Year year;
    private int week;
    private double avgHumidity;
    private double avgLight;
    private double avgSoilMoisture;
    private double avgTemperatureAsc;
    private double avgTemperatureAsf;
    private double avgWaterLevel;
}

File: Greenhouse-FrontEnd/src/app/models/GreenhouseWeeklyAverageDTO.ts
export class GreenhouseWeeklyAverageDTO {
  year: string;
  week: number;
  avgHumidity: number;
  avgLight: number;
  avgSoilMoisture: number;
  avgTemperatureAsc: number;
  avgTemperatureAsf: number;
  avgWaterLevel: number;

  constructor(year: string, week: number, avgHumidity: number, avgLight: number, avgSoilMoisture: number, avgTemperatureAsc: number, avgTemperatureAsf: number, avgWaterLevel: number) {
    this.year = year;
    this.week = week;
    this.avgHumidity = avgHumidity;
    this.avgLight = avgLight;
    this.avgSoilMoisture = avgSoilMoisture;
    this.avgTemperatureAsc = avgTemperatureAsc;
    this.avgTemperatureAsf = avgTemperatureAsf;
    this.avgWaterLevel = avgWaterLevel;
  }
}
